### ISSUE TITLE
a11y: Project management collaboration table

### ISSUE THREAD

**Issue by woodruffw** [pypa/warehouse#6077]
Minor accessibility improvements can be made to the collaboration table on `/manage/project/{project_name}/collaboration/`:

* The `<table>` itself should have a `<caption>`, as the first child node. This is partially mitigated by the `<h2>` above
* The `<table>` should include `scope=col` or another appropriate scope for each `<th>` node
* Each user's name and icon can be combined into a single clickable link, and the icon can be set to `aria-hidden=true` (alt text is currently provided, but duplicates the information provided by the plain text right next to the icon)
